coupled_laplace_problem/coupled_laplace_problem.cc

```diff
@@ -0,0 +1,674 @@
+            cell_rhs(i) += (fe_values.shape_value(i, q_index) * // phi_i(x_q)
+                            right_hand_side.value(x_q) *        // f(x_q)
+                            fe_values.JxW(q_index));            // dx
+          }
+
+      cell->get_dof_indices(local_dof_indices);
+      for (const unsigned int i : fe_values.dof_indices())
+        {
+          for (const unsigned int j : fe_values.dof_indices())
+            system_matrix.add(local_dof_indices[i],
+                              local_dof_indices[j],
+                              cell_matrix(i, j));
+
+          system_rhs(local_dof_indices[i]) += cell_rhs(i);
+        }
+    }
+  {
+    // At first, we apply the Dirichlet boundary condition from step-4, as
+    // usual.
+    std::map<types::global_dof_index, double> boundary_values;
+    VectorTools::interpolate_boundary_values(dof_handler,
+                                             0,
+                                             BoundaryValues<dim>(),
+                                             boundary_values);
+    MatrixTools::apply_boundary_values(boundary_values,
+                                       system_matrix,
+                                       solution,
+                                       system_rhs);
+  }
+  {
+    // Afterwards, we apply the coupling boundary condition. The `boundary_data`
+    // has already been filled by preCICE.
+    MatrixTools::apply_boundary_values(boundary_data,
+                                       system_matrix,
+                                       solution,
+                                       system_rhs);
+  }
+}
+
+
+
+template <int dim>
+void
+LaplaceProblem<dim>::solve()
+{
+  SolverControl            solver_control(1000, 1e-12);
+  SolverCG<Vector<double>> solver(solver_control);
+  solver.solve(system_matrix, solution, system_rhs, PreconditionIdentity());
+
+  std::cout << "   " << solver_control.last_step()
+            << " CG iterations needed to obtain convergence." << std::endl;
+}
+
+
+
+template <int dim>
+void
+LaplaceProblem<dim>::output_results() const
+{
+  DataOut<dim> data_out;
+
+  data_out.attach_dof_handler(dof_handler);
+  data_out.add_data_vector(solution, "solution");
+
+  data_out.build_patches();
+
+  std::ofstream output(dim == 2 ? "solution-2d.vtk" : "solution-3d.vtk");
+  data_out.write_vtk(output);
+}
+
+
+
+template <int dim>
+void
+LaplaceProblem<dim>::run()
+{
+  std::cout << "Solving problem in " << dim << " space dimensions."
+            << std::endl;
+
+  make_grid();
+  setup_system();
+
+  // After we set up out system, we initialize preCICE using the functionalities
+  // of our Adapter.
+  adapter.initialize(dof_handler, boundary_data);
+
+  // preCICE steers the coupled simulation completely. The steering is provided
+  // by the function `isCouplingOngoing`, which tells the solver, whether the
+  // time loop has already been finished or not. This tutorial solves 'just' an
```

Comment:
Could we make that shorter?
`isCouplingOngoing` is only to synchronize the end of the simulation with the coupling partner. Not for steering in general (at least in the preCICE documentation, we use the word "steering" differently).